### PR TITLE
Rename hooks

### PR DESCRIPTION
**Pull request by cloudkucooland**
try this again, to the correct branch

### CODE REVIEW

**Comment by le-jeu** [comment]
missed: `init.js:182` `window.map.fire("wasabeeDkeys", { reason: "startup" }, false);`

**Comment by le-jeu** [comment]
current hooks are:
```
wasabee:crosslinks
wasabee:crosslinks:done
wasabee:defensivekeys
wasabee:op:add
wasabee:op:background
wasabee:op:delete
wasabee:op:select
wasabee:op:showhide
wasabee:paneclear
wasabee:paneset
wasabee:uiupdate
```

**Comment by cloudkucooland** [comment]
Can we split map data updates from other ui updates? We do not need to redraw all buttons on map data change

**Comment by cloudkucooland** [comment]
wasabee:uiupdate is replaced by

wasabee:uiupdate:mapdata (for changes that need to draw the map or dialogs of map data)
wasabee:uiupdate:teamdata (for dialogs that display team info)
wasabee:uiupdate:buttons (to update the buttons)
wasabee:ui:buttonreset (to completely reset the buttons on skin change)
wasabee:uiupdate:agentlocations (to update agents on the map when they move)
wasabee:uiupdate:settings (for the settings dialog which doesn't need to change on map/team/agentlocation changes)

I think that's all.

This builds and is ready for review.

File: src/code/skin.js [hunk 1]
```diff
@@ -106,8 +104,8 @@ export function changeSkin(names) {
   localStorage[window.plugin.wasabee.static.constants.SKIN_KEY] =
     JSON.stringify(validNames);
 
-  addButtons();
-  window.map.fire("wasabeeUIUpdate", { reason: "skin change" }, false);
+  window.map.fire("wasabee:ui:buttonreset");
+  window.map.fire("wasabee:uiupdate:mapdata", { reason: "skin change" }, false); // XXX is this needed?
```
Comment (le-jeu) [inline src/code/skin.js:108]: skin also concerns marker icons and link style

File: src/code/buttons/quickdrawButton.js [hunk 2]
```diff
@@ -137,7 +137,7 @@ const QuickDrawControl = L.Handler.extend({
     window.addHook("portalSelected", this._portalClickedHook);
 
     // Leaflet format for leaflet DOM event
-    this._map.on("wasabeeUIUpdate", this._uiupdate, this);
+    this._map.on("wasabee:uiupdate:mapdata", this._uiupdate, this);
```
Comment (le-jeu) [inline src/code/buttons/quickdrawButton.js:140]: why not use `wasabee:op:select` for such ui update ?

File: src/code/dialogs/blockersList.js [hunk 3]
```diff
@@ -45,7 +45,11 @@ const BlockerList = WDialog.extend({
     const buttons = {};
     buttons[wX("OK")] = () => {
       this.closeDialog();
-      window.map.fire("wasabeeUIUpdate", { reason: "blockerlist" }, false);
```
Comment (le-jeu) [inline src/code/dialogs/blockersList.js:48]: this line looked like a wrong copy/paste

File: src/code/dialogs/confirmDialog.js [hunk 4]
```diff
@@ -46,7 +46,11 @@ const ConfirmDialog = WDialog.extend({
 
   removeHooks: function () {
     WDialog.prototype.removeHooks.call(this);
-    window.map.fire("wasabeeUIUpdate", { reason: "confirmDialog" }, false);
```
Comment (le-jeu) [inline src/code/dialogs/confirmDialog.js:49]: should this be handled by the `callback` instead ?

File: src/code/dialogs/importDialog.js [hunk 5]
```diff
@@ -26,8 +26,8 @@ const ImportDialog = WDialog.extend({
   removeHooks: function () {
     WDialog.prototype.removeHooks.call(this);
 
-    window.map.fire("wasabeeUIUpdate", { reason: "import" }, false);
-    window.map.fire("wasabeeCrosslinks", { reason: "import" }, false);
+    window.map.fire("wasabee:uiupdate:mapdata", { reason: "import" }, false);
```
Comment (le-jeu) [inline src/code/dialogs/importDialog.js:29]: this can also be moved to the import part instead of `removeHooks`

**Comment by le-jeu** [comment]
> wasabee:uiupdate is replaced by
> 
> wasabee:uiupdate:mapdata (for changes that need to draw the map or dialogs of map data)
> wasabee:uiupdate:teamdata (for dialogs that display team info)
> wasabee:uiupdate:buttons (to update the buttons)
> wasabee:ui:buttonreset (to completely reset the buttons on skin change)
> wasabee:uiupdate:agentlocations (to update agents on the map when they move)
> wasabee:uiupdate:settings (for the settings dialog which doesn't need to change on map/team/agentlocation changes)
> 
> I think that's all.
> 
> This builds and is ready for review.

Maybe reverse the logic, with `wasabee:uiupdate:*` we fire __actions__ , could it be clearer to use __events__ instead ? 
instead of wondering if the UI need to update when synching OPs, let the UI register for `op/team:update`-like events

File: src/code/buttons/quickdrawButton.js (hunk 2 above)
Comment (cloudkucooland) [reply src/code/buttons/quickdrawButton.js:140]: yes, this not redrawing on all mapdata change is going to help.

File: src/code/dialogs/blockersList.js (hunk 3 above)
Comment (cloudkucooland) [reply src/code/dialogs/blockersList.js:48]: There was a reason I added this, but I can't recall now.

File: src/code/dialogs/confirmDialog.js (hunk 4 above)
Comment (cloudkucooland) [reply src/code/dialogs/confirmDialog.js:49]: yes, did a sweep to make this change

File: src/code/dialogs/importDialog.js (hunk 5 above)
Comment (cloudkucooland) [reply src/code/dialogs/importDialog.js:29]: yes

File: src/code/dialogs/promptDialog.js [hunk 6]
```diff
@@ -25,7 +25,13 @@ const PromptDialog = WDialog.extend({
 
   removeHooks: function () {
     WDialog.prototype.removeHooks.call(this);
-    window.map.fire("wasabeeUIUpdate", { reason: "PromptDialogClose" }, false);
+    // could be either -- maybe more
+    window.map.fire("wasabee:uiupdate:teamdata");
```
Comment (cloudkucooland) [inline src/code/dialogs/promptDialog.js:29]: These need to be moved to the callbacks instead.

**Review by le-jeu — APPROVED** [review]

**Pull request merged**